x/exchange/msg.go

```diff
@@ -0,0 +1,464 @@
+	addr := sdk.MustAccAddressFromBech32(m.Buyer)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketSettleRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	if err := ValidateOrderIDs("ask", m.AskOrderIds); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := ValidateOrderIDs("bid", m.BidOrderIds); err != nil {
+		errs = append(errs, err)
+	}
+
+	inBoth := IntersectionUint64(m.AskOrderIds, m.BidOrderIds)
+	if len(inBoth) > 0 {
+		errs = append(errs, fmt.Errorf("order ids duplicated as both bid and ask: %v", inBoth))
+	}
+
+	// Nothing to validate now for the ExpectPartial flag.
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketSettleRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketWithdrawRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, errors.New("invalid market id: cannot be zero"))
+	}
+
+	if _, err := sdk.AccAddressFromBech32(m.ToAddress); err != nil {
+		errs = append(errs, fmt.Errorf("invalid to address %q: %w", m.ToAddress, err))
+	}
+
+	if err := m.Amount.Validate(); err != nil {
+		errs = append(errs, fmt.Errorf("invalid amount %q: %w", m.Amount, err))
+	} else if m.Amount.IsZero() {
+		errs = append(errs, fmt.Errorf("invalid amount %q: cannot be zero", m.Amount))
+	}
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketWithdrawRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketUpdateDetailsRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	if err := m.MarketDetails.Validate(); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketUpdateDetailsRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketUpdateEnabledRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	// Nothing to validate for the AcceptingOrders field.
```

Comment:
We can probably get rid of this comment.

Reply:
It's nice to have it so that it's clear that validation on that field wasn't simply forgotten.

Reply:
I don't think these are necessary.  The worst case is that a developer will look into it themselves and discover it isn't needed.  However, there may be a future change that could require a validation to happen to a field that wasn't required in the previously this would then cause a little confusion.

Reply:
Why is it bad to have it here? Why is it bad to save a developer the trouble of re-investigating something?
If there's a future change that could require validation to happen, they'd be looking at stuff related to the field anyway, knowing there's a change and would know immediately that the comment was no longer accurate.

Reply:
The main reason why I suggested getting rid of this comment is because it's self explanatory. This is the value we are updating and there are only two acceptable values. The only subset of this is one value, and if that were the case the boolean would not be included. The message would either be titled with Enable or Disable. I think comments are great, but I think we get the idea from the message and its contents.

Reply:
But it's not self explanatory. In order to know that there isn't any validation involving that field, you have to have intimate knowledge of the containing struct and how it should behave.